src/coreclr/classlibnative/float/floatdouble.cpp

```diff
@@ -280,6 +280,21 @@ FCIMPL1_V(double, COMDouble::Sin, double x)
     return (double)sin(x);
 FCIMPLEND
 
+/*====================================SinCos====================================
+**
+==============================================================================*/
+FCIMPL3_VII(void, COMDouble::SinCos, double x, double* pSin, double* pCos)
+    FCALL_CONTRACT;
+
+#ifdef _MSC_VER
+    *pSin = (double)sin(x);
+    *pCos = (double)cos(x);
```

Comment:
```suggestion
    *pSin = sin(x);
    *pCos = cos(x);
```
?

Reply:
All of the `COMDouble` and `COMSingle` APIs convert the return values. I think this is largely due to legacy reasons at this point, but all new additions have done it as well for consistency.

Reply:
@jkotas if this is something we'd like to change, I can do it here for `sincos` and grab everything else in a follow up PR?

Reply:
Sounds good to me. I have not realized that you are just replicating the existing unnecessary casts.